test/SparqlAntlrParserTest.cpp

```diff
@@ -661,3 +661,120 @@ TEST(SparqlParser, Bind) {
     EXPECT_THAT(bindAndText._remainingText, IsEmpty());
   }
 }
+
+TEST(SparqlParser, Integer) {
+  {
+    string input = "1931";
+    ParserAndVisitor p{input};
+
+    unsigned long long result =
+        p.parser.integer()->accept(&p.visitor).as<unsigned long long>();
+    EXPECT_EQ(result, 1931ULL);
+  }
+
+  {
+    string input = "0";
+    ParserAndVisitor p{input};
+
+    unsigned long long result =
+        p.parser.integer()->accept(&p.visitor).as<unsigned long long>();
+    EXPECT_EQ(result, 0ULL);
+  }
+
+  {
+    string input = "18446744073709551615";
+    ParserAndVisitor p{input};
+
+    unsigned long long result =
+        p.parser.integer()->accept(&p.visitor).as<unsigned long long>();
+    EXPECT_EQ(result, 18446744073709551615ULL);
+  }
+
+  {
+    string input = "18446744073709551616";
+    ParserAndVisitor p{input};
+
+    EXPECT_THROW(p.parser.integer()->accept(&p.visitor), SparqlParseException);
+  }
+
+  {
+    string input = "10000000000000000000000000000000000000000";
+    ParserAndVisitor p{input};
+
+    EXPECT_THROW(p.parser.integer()->accept(&p.visitor), SparqlParseException);
+  }
+
+  {
+    string input = "-1";
+    ParserAndVisitor p{input};
+
+    EXPECT_THROW(p.parser.integer()->accept(&p.visitor),
+                 antlr4::ParseCancellationException);
+  }
+}
+
+TEST(SparqlParser, LimitOffsetClause) {
+  {
+    string input = "LIMIT 10";
+
+    sparqlParserHelpers::ResultOfParseAndRemainingText<LimitOffsetClause>
+        limitOffset = sparqlParserHelpers::parseLimitOffsetClause(input, {});
+
+    EXPECT_THAT(limitOffset._resultOfParse, IsLimitOffset(10ULL, 1ULL, 0ULL));
+    EXPECT_THAT(limitOffset._remainingText, IsEmpty());
+  }
+
+  {
+    string input = "OFFSET 31 LIMIT 12 TEXTLIMIT 14";
+
+    sparqlParserHelpers::ResultOfParseAndRemainingText<LimitOffsetClause>
+        limitOffset = sparqlParserHelpers::parseLimitOffsetClause(input, {});
+
+    EXPECT_THAT(limitOffset._resultOfParse, IsLimitOffset(12ULL, 14ULL, 31ULL));
+    EXPECT_THAT(limitOffset._remainingText, IsEmpty());
```

Comment:
It would be nice to make this a 
"ExpectCompletelyParsed(limitOffset, IsLimitOffset(....));" because the `IsEmpty` check is really redundant and repetitive, but I think that can be a different PR, and maybe I should check, how to implement something like that with Google matchers.

Reply:
In fact it is as simple as this, so it can be in this PR.:

```
void expectCompleteParse(const auto& resultOfParseAndText, auto&& matcher) {
  EXPECT_THAT(resultOfParseAndText._resultOfParse, matcher);
  EXPECT_TRUE(resultOfParseAndText._remainingText.empty());
}
```
Usage:
`expectCompleteParse(limitOffset, IsLimitOffset(12u, 14u, 31u));`

I delivered the code, so you can add the documentation as for what it does:)

Reply:
Is there a preferred docstring format? I have found docstrings in javadoc style and also other formats in the codebase.

Reply:
In general we try to use the doxygen syntax with /// for short docstrings and /** for longer comments.
Although we currently do not build Doxygen documentations regularly at the moment, it would be nice to leave the possibility.